compiler/modules/front/codeGen.bal

```diff
@@ -983,16 +988,18 @@ function codeGenCompoundAssignToVar(CodeGenContext cx,
                                     s:Expr rexpr,
                                     s:BinaryArithmeticOp|s:BinaryBitwiseOp op,
                                     Position pos) returns CodeGenError|StmtEffect {
-    s:Expr expr;
-    Position startPos = rexpr.startPos;
-    Position endPos = rexpr.endPos;
-    if op is s:BinaryArithmeticOp {
-        expr = { startPos, endPos, opPos: pos, arithmeticOp: op, left: lValue, right: rexpr };
+    var { block: nextBlock, result: lhs } = check codeGenExpr(cx, startBlock, env, lValue);
```

Comment:
Don't understand what you're doing here.  lookupVarRefForAssign will give you the register.  So why can't you use that?

The codeGenExprForInt in the codeGenCompoundAssignToMember is for the index expression: there's nothing like that here.

Reply:
fixed